packages/utils/src/envelope.ts

```diff
@@ -41,16 +40,32 @@ export function addItemToEnvelope<E extends Envelope>(envelope: E, newItem: E[1]
 /**
  * Convenience function to loop through the items and item types of an envelope.
  * (This function was mostly created because working with envelope types is painful at the moment)
+ *
+ * If the callback returns true, the rest of the items will be skipped.
  */
 export function forEachEnvelopeItem<E extends Envelope>(
   envelope: Envelope,
-  callback: (envelopeItem: E[1][number], envelopeItemType: E[1][number][0]['type']) => void,
-): void {
+  callback: (envelopeItem: E[1][number], envelopeItemType: E[1][number][0]['type']) => boolean | void,
+): boolean {
   const envelopeItems = envelope[1];
-  envelopeItems.forEach((envelopeItem: EnvelopeItem) => {
+
+  for (const envelopeItem of envelopeItems) {
     const envelopeItemType = envelopeItem[0].type;
-    callback(envelopeItem, envelopeItemType);
-  });
+    const result = callback(envelopeItem, envelopeItemType);
+
+    if (result) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/**
+ * Returns true if the envelope contains any of the given envelope item types
+ */
+export function envelopeContainsItemType(envelope: Envelope, ...types: EnvelopeItemType[]): boolean {
```

Comment:
l: I think I would _slightly_ prefer to pass the types as array (`types: EnvelopeItem[]`). A bit easier if we ever want to refactor/extend this somehow, and since we only use this internally I think it's also OK to expect that for a theoretical singular type (which we don't even have right now). But no strong feelings here, if you feel it is better this way, feel free to leave it.

Reply:
Ah, I agree that spread arguments are not great. Too easy to pass zero `types` and a host of other usability issues!